brainscore/submission/evaluation.py

```diff
@@ -24,47 +22,64 @@
-            function = lambda: module.get_model(model)
-            base_model_pool[model] = LazyLoad(function)
-            try:
-                layers[model] = module.get_layers(model)
-            except Exception:
-                logging.warning(f'Could not retrieve layer for model {model} -- skipping model')
-        model_layers = ModelLayers(layers)
-        ml_brain_pool = MLBrainPool(base_model_pool, model_layers)
+    if isinstance(submission_config, MultiConfig):
+        # We rerun existing models, which potentially are defined in different submissions
+        for submission in submission_config.submissions.values():
+            module = prepare_module(submission, submission_config)
+            models = []
+            for model in submission_config.models:
+                if model.submission.id == submission.id:
+                    models.append(model)
+            assert len(models) > 0
+            run_submission(module, models, test_benchmarks, submission, submission_config.jenkins_id)
     else:
-        logger.info(f"Start working with brain models")
-        for model in test_models:
-            ml_brain_pool[model] = module.get_model(model)
+        submission = submission_config.submission
+        try:
+            module = prepare_module(submission, submission_config)
+            test_models = module.get_model_list() if models is None or len(models) == 0 else models
+            assert len(test_models) > 0
+            model_instances = []
+            for model in test_models:
+                reference = None
+                if hasattr(module, 'get_bibtex'):
+                    bibtex_string = module.get_bibtex(model)
+                    reference = get_reference(bibtex_string)
+                model_instances.append(
+                    Model.create(name=model, owner=submission.submitter, public=submission_config.public,
+                                 reference=reference, submission=submission))
+            run_submission(module, model_instances, test_benchmarks, submission, submission.id)
+        except Exception as e:
+            submission.status = 'failure'
+            submission.save()
+            logging.error(f'Could not install submission because of following error')
+            logging.error(e, exc_info=True)
+            raise e
+
+
+def run_submission(module, test_models, test_benchmarks, submission, jenkins_id):
+    ml_brain_pool = get_ml_pool(test_models, module, submission)
     data = []
+    success = True
     try:
-        for model_id in test_models:
+        for modelInst in test_models:
+            model_id = modelInst.name
             for benchmark in test_benchmarks:
+                scoreInst = None
                 try:
+                    start = datetime.datetime.now()
+                    bench_inst = get_benchmark_instance(benchmark)
+                    scoreInst = Score.create(benchmark=bench_inst, start_timestamp=start, model=modelInst)
```

Comment:
there's some inconsistent naming here (`bench_inst` vs `scoreInst`), would prefer `benchmark_instance` and `score`

Reply:
ok I tried to make it more transparent. Database objects are called `_entry`, strings are called `_name` and brainscore objects keep the base name